src/shared/credentials/userCredentialsUtils.ts

```diff
@@ -57,11 +57,18 @@ export class UserCredentialsUtils {
      *
      * @returns array of filenames for files found.
      */
-    public static async findExistingCredentialsFilenames(): Promise<string[]> {
+    public static async findExistingCredentialsFilenames(isCredentialsRequired = false): Promise<string[]> {
         const candidateFiles: string[] = [getCredentialsFilename(), getConfigFilename()]
 
         const existsResults: boolean[] = await Promise.all(
-            candidateFiles.map(async filename => await SystemUtilities.fileExists(filename))
+            candidateFiles.map(async filename => {
+                const exists = await SystemUtilities.fileExists(filename)
+                if (!exists || !isCredentialsRequired) {
+                    return exists
+                }
+                const contents = await SystemUtilities.readFile(filename)
+                return /aws_access_key_id|aws_secret_access_key|aws_session_token/i.test(contents)
```

Comment:
This is rather fragile. We already have code that logs things like this:

```
2022-11-02 22:01:54 [ERROR]: Profile foo is not a valid Credential Profile: not supported by the Toolkit
2022-11-02 22:01:54 [WARN]: Shared Credentials Profile foo is not valid. It will not be used by the toolkit.
```

can we use that logic as the source of truth?